docs/hugo/content/architecture-decision-records/ADR-2022-02-Adoption-Policy.md

```diff
@@ -0,0 +1,127 @@
+   This is useful for scenarios such as moving ownership of resources from ASO in cluster A to ASO in cluster B.
+   This is enabled by `reconcile-policy`.
+
+### Open question: what should the default `reconcile-policy` be for resources which are adopted?
+
+There are two main options.
+
+#### Option 1: The default `reconcile-policy` for adopted resources is the same as for resources ASO created.
+
+ASO automatically adopts, updates, and will delete backing Azure resources as normal unless the user
+explicitly sets `reconcile-policy`.
+
+**Advantages:**
+- It's easy to explain what the behavior is. Simplistic users don't have to understand `reconcile-policy` (they won't
+  ever see it unless they set it themselves).
+
+**Disadvantages:**
+- Users have no good way to know their resource existed previously. They may `kubectl delete` it without realizing that
+  other things depend on this resource. Deletion (and some mutation) is destructive in Azure so lost data, etc cannot 
+  easily be recovered.
+
+**Thoughts/comments:**
+
+This sort-of matches the behavior of ARM templates and SDKs. If you call the `DELETE` API the resource is gone.
+
+Some discussion of this topic has happened over in [ACK](https://github.com/aws-controllers-k8s/community/issues/82).
+
+[Crossplane](https://github.com/crossplane/crossplane/issues/2590) doesn't support any notion of `reconcile-policy` at all.
+`kubectl delete` always deletes the backing resource no matter what.
+
+[GCP](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources#creating_a_resource) does it this way:
+> When you create a resource, Config Connector creates the resource if it doesn't exist. If a Google Cloud resource 
+> already exists with the same name, then Config Connector acquires the resource and manages it.
+
+GCP also [supports](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources) a 
+`reconcile-policy`-like annotation, so doing it like this would basically be doing it exactly like GCP.
+
+Kuberentes has a similar concept for `StorageClasses`/`PersistentVolumes` called 
+[reclaimPolicy](https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy), whose default
+seems to be `Delete`.
+
+#### Option 2: The default `reconcile-policy` for adopted resource is `abandon-on-delete`.
+
+ASO detects that a resource previously existed and sets `reconcile-policy` automatically to `abandon-on-delete`.
+A new option in the ASO config `adoption-policy` would control this default so users could opt into the Option 1 style
+discussed above if they wanted (but it's not the default behavior).
+
+**Advantages:**
+- More flexibility. Users can choose which behavior they want when deploying ASO. 
+- Defaults to a more conservative `reconcile-policy` to protect them from accidentally deleting things they may not
+  have wanted to delete.
+
+**Disadvantages:**
```

Comment:
I think we can add one more to this list which is potential inconsistency across resources.
Imagine you have a helm chart of 10 ASO resources, 5 of which were already present on Azure, and the 5 others are just introduced through the helm chart. Now half is 'owned' by the ASO controller and the other half is not. You forget about that and a long time later you run `helm uninstall` because you intend to get rid of all the resources in the chart. And in fact even all the resources get deleted from the cluster and you won't see them anymore. But half of those resources would still continue to remain in Azure.

Reply:
I covered this I think with this line:
> Bias towards safety may cause resource user thinks was deleted to continue existing (and billing) in Azure. Not a data
  loss though.